app/tally_handler.go

```diff
@@ -0,0 +1,253 @@
+	results := make(map[types.VoteOption]sdk.Dec)
+	results[types.OptionYes] = sdk.ZeroDec()
+	results[types.OptionAbstain] = sdk.ZeroDec()
+	results[types.OptionNo] = sdk.ZeroDec()
+	results[types.OptionNoWithVeto] = sdk.ZeroDec()
+
+	totalVotingPower := sdk.ZeroDec()
+	currValidators := make(map[string]types.ValidatorGovInfo)
+
+	// fetch all the bonded validators, insert them into currValidators
+	th.stk.IterateBondedValidatorsByPower(ctx, func(index int64, validator stakingtypes.ValidatorI) (stop bool) {
+		currValidators[validator.GetOperator().String()] = types.NewValidatorGovInfo(
+			validator.GetOperator(),
+			validator.GetBondedTokens(),
+			validator.GetDelegatorShares(),
+			sdk.ZeroDec(),
+			types.WeightedVoteOptions{},
+		)
+
+		return false
+	})
+
+	th.gk.IterateVotes(ctx, proposal.ProposalId, func(vote types.Vote) bool {
+		// if validator, just record it in the map
+		voter, err := sdk.AccAddressFromBech32(vote.Voter)
+
+		if err != nil {
+			panic(err)
+		}
+
+		valAddrStr := sdk.ValAddress(voter.Bytes()).String()
+		if val, ok := currValidators[valAddrStr]; ok {
+			val.Vote = vote.Options
+			currValidators[valAddrStr] = val
+		}
+
+		// iterate over all delegations from voter, deduct from any delegated-to validators
+		th.stk.IterateDelegations(ctx, voter, func(index int64, delegation stakingtypes.DelegationI) (stop bool) {
+			valAddrStr := delegation.GetValidatorAddr().String()
+
+			if val, ok := currValidators[valAddrStr]; ok {
+				// There is no need to handle the special case that validator address equal to voter address.
+				// Because voter's voting power will tally again even if there will deduct voter's voting power from validator.
+				val.DelegatorDeductions = val.DelegatorDeductions.Add(delegation.GetShares())
+				currValidators[valAddrStr] = val
+
+				// delegation shares * bonded / total shares
+				votingPower := delegation.GetShares().MulInt(val.BondedTokens).Quo(val.DelegatorShares)
+
+				for _, option := range vote.Options {
+					subPower := votingPower.Mul(option.Weight)
+					results[option.Option] = results[option.Option].Add(subPower)
+				}
+				totalVotingPower = totalVotingPower.Add(votingPower)
+			}
+
+			return false
+		})
+
+		// get voter bkava and update total voting power and results
+		addrBkava := th.getAddrBkava(ctx, voter).toCoins()
+		for _, coin := range addrBkava {
+			valAddr, err := liquidtypes.ParseLiquidStakingTokenDenom(coin.Denom)
+			if err != nil {
+				break
+			}
+
+			// reduce delegator shares by the amount of voter bkava for the validator
+			valAddrStr := valAddr.String()
+			if val, ok := currValidators[valAddrStr]; ok {
+				val.DelegatorDeductions = val.DelegatorDeductions.Add(coin.Amount.ToDec())
+				currValidators[valAddrStr] = val
+			}
+
+			// votingPower = amount of ukava coin
+			votingPowerInt, err := th.lk.GetKavaForDerivatives(ctx, sdk.NewCoins(coin))
```

Comment:
In master this was updated to `GetStakedTokensForDerivatives` and returns `sdk.Coin`

Reply:
:+1:, updated to reflect this.